app/src/main/java/com/dragic/gamehunter/viewmodel/GameDetailsViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.dragic.gamehunter.viewmodel
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.dragic.gamehunter.di.BG_DISPATCHER
+import com.dragic.gamehunter.di.GameId
+import com.dragic.gamehunter.repository.DealRepository
+import com.dragic.gamehunter.utils.toDealDetailsViewState
+import com.dragic.gamehunter.utils.toImageContentViewState
+import com.dragic.gamehunter.view.gamedetails.DealDetailsViewState
+import com.dragic.gamehunter.view.gamedetails.ImageContentViewState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Named
+import kotlin.coroutines.CoroutineContext
+
+const val DEAL_REDIRECT_LINK_PREFIX = "https://www.cheapshark.com/redirect?dealID="
```

Comment:
This should get mapped when you map the response to the entity. The ViewModel should not concern itself with this